src/Tasks/WriteCodeFragment.cs

```diff
@@ -110,6 +110,12 @@ public override bool Execute()
                 {
                     OutputFile = new TaskItem(Path.Combine(OutputDirectory.ItemSpec, OutputFile.ItemSpec));
                 }
+                else if (OutputFile != null && OutputDirectory == null && !Path.IsPathRooted(OutputFile.ItemSpec))
+                {
+                    // In case specified OutputFile path is a file name but OutputDirectory is not set, expand it to a fully qualified path based on current directory.
+                    // So that it has the directory given to ensuring directory exists.
+                    OutputFile.ItemSpec = Path.GetFullPath(OutputFile.ItemSpec);
```

Comment:
Modifying `OutputFile.ItemSpec` feels wrong to me. Could you instead either

1. Use a local variable for "full path of file", or
2. When doing filesystem operations, use the item's `Directory` and `FullPath` metadata rather than computing it directly?

Reply:
`Directory` metadata returns full path of item directory relative to root. So I had to use `FullPath` metadata.